pinot-spi/src/main/java/org/apache/pinot/spi/env/CommonsConfigurationUtils.java

```diff
@@ -74,19 +65,19 @@ public static PropertiesConfiguration fromInputStream(InputStream stream)
    */
   public static PropertiesConfiguration fromPath(String path)
       throws ConfigurationException {
-    return fromPath(path, false, true);
+    return fromPath(path, true, PropertyIOFactoryKind.DefaultPropertyConfigurationIOFactory);
   }
 
   /**
    * Instantiate a {@link PropertiesConfiguration} from an {@link String}.
    * @param path representing the path of file
-   * @param setIOFactory representing to set the IOFactory or not
    * @param setDefaultDelimiter representing to set the default list delimiter.
    * @return a {@link PropertiesConfiguration} instance.
    */
-  public static PropertiesConfiguration fromPath(String path, boolean setIOFactory, boolean setDefaultDelimiter)
+  public static PropertiesConfiguration fromPath(String path, boolean setDefaultDelimiter,
+      PropertyIOFactoryKind ioFactoryKind)
```

Comment:
how about pass in `@Nullable IOFactory ioFactory` to those util methods? so that we decouple those util methods from the logic of deciding the type and initializing the ioFactory object.